Update balancer strategy recommendations (#11759)
apache/druid: docs/configuration/index.md

|`druid.coordinator.kill.period`|How often to send kill tasks to the indexing service. Value must be greater than `druid.coordinator.period.indexingPeriod`. Only applies if kill is turned on.|P1D (1 Day)|
|`druid.coordinator.kill.durationToRetain`| Do not kill unused segments in last `durationToRetain`, must be greater or equal to 0. Only applies and MUST be specified if kill is turned on. Note that default value is invalid.|PT-1S (-1 seconds)|
|`druid.coordinator.kill.maxSegments`|Kill at most n unused segments per kill task submission, must be greater than 0. Only applies and MUST be specified if kill is turned on. Note that default value is invalid.|0|

> |`druid.coordinator.balancer.strategy`|Specify the type of balancing strategy that the coordinator should use to distribute segments among the historicals. `cachingCost` is logically equivalent to `cost` but is more CPU-efficient on large clusters.|`cost`|

Reviewer: ```suggestion
|`druid.coordinator.balancer.strategy`|Specify the type of balancing strategy for the coordinator to use to distribute segments among the historicals. `cachingCost` is logically equivalent to `cost` but is more CPU-efficient on large clusters.|`cost`|
```
nit: should sounds conditional

Reviewer 2: diskNormalized and random still exist as options - so we should still mention them in the docs, but it would be good to note the pitfalls of using diskNormalized and random.

Perhaps in a future change, we can get rid of them, if we think cost and cachingCost are better options. I'll need to look through some notes to see what the drawbacks were with diskNormalized

Reviewer 2: @techdocsmith - do you have a better suggestion for mentioning the other 2 strategies and their pitfalls?

```suggestion
|`druid.coordinator.balancer.strategy`|Specify the type of balancing strategy for the coordinator to use to distribute segments among the historicals. `cachingCost` is logically equivalent to `cost` but is more CPU-efficient on large clusters. `diskNormalized` weights the costs according to the servers disk usage ratios - there are known issues with this strategy distributing segments unevenly across the cluster. `random` will randomly pick processes to distribute segments.|`cost`|
```

Reviewer: I think this is fine. It just highlights a limitation to trying to describe the different options w/in a table cell. Small nit: `random` picks processes. Otherwise this works.

> |`druid.coordinator.balancer.strategy`|Specify the type of balancing strategy for the coordinator to use to distribute segments among the historicals. `cachingCost` is logically equivalent to `cost` but is more CPU-efficient on large clusters. `diskNormalized` weights the costs according to the servers disk usage ratios - there are known issues with this strategy distributing segments unevenly across the cluster. `random` distributes segments among services randomly.|`cost`|

Reviewer 2: ```suggestion
|`druid.coordinator.balancer.strategy`|Specify the type of balancing strategy for the coordinator to use to distribute segments among the historicals. `cachingCost` is logically equivalent to `cost` but is more CPU-efficient on large clusters. `diskNormalized` weights the costs according to the servers' disk usage ratios - there are known issues with this strategy distributing segments unevenly across the cluster. `random` distributes segments among services randomly.|`cost`|
```
